### ISSUE TITLE
Make basename optional

### ISSUE THREAD

**Issue by blueyed** [so-fancy/diff-so-fancy#280]
I prefer not having `basename` called inside the hunk - since this is where the line number is appended, and therefore is good for copy'n'pasting into your editor.

Currently there is `foo:42`, but the original would be `bar/foo:42`.

```diff
diff --git i/diff-so-fancy w/diff-so-fancy
index bece8e0..cd1397a 100755
--- i/diff-so-fancy
+++ w/diff-so-fancy
@@ -241,7 +241,7 @@ sub do_dsf_stuff {
                        }
 
                        my ($orig_offset, $orig_count, $new_offset, $new_count) = parse_hunk_
-                       $last_file_seen = basename($last_file_seen);
+                       # $last_file_seen = basename($last_file_seen);
 
                        # Figure out the start line
                        my $start_line = start_line_calc($new_offset,$new_count);
```
Related discussion: https://github.com/so-fancy/diff-so-fancy/issues/100#issuecomment-192040721

**Comment by scottchiefbaker** [comment]
Can you give me a real world example of where/how this would change the d-s-f output?

**Comment by blueyed** [comment]
Any diff where the file is in a subdirectory.

**Comment by scottchiefbaker** [comment]
Ahhh I see... Honestly I don't know why we did it this way at all. It probably should just be the full path in the hunk line.

Should we make this the default? Or make it an optional setting?

**Comment by blueyed** [comment]
I vote for making it the default.
/cc @paulirish

**Comment by scottchiefbaker** [comment]
For what it's worth, I think my vote is making it default as well.

**Comment by paulirish** [comment]
yeah seems good. let's do the full path.

**Comment by scottchiefbaker** [comment]
Ok this should be fixed in 49969f04db747138243d598c43d766c7e65c1f27. If there is anything else on this let me know and I'll reopen.
